lib/spack/spack/binary_distribution.py

```diff
@@ -736,7 +736,7 @@ def get_buildinfo_dict(spec):
         "relocate_binaries": manifest["binary_to_relocate"],
         "relocate_links": manifest["link_to_relocate"],
         "hardlinks_deduped": manifest["hardlinks_deduped"],
-        "hash_to_prefix": hashes_to_prefixes(spec),
+        "hash_to_prefix": {h: info[1] for h, info in hashes_to_prefixes(spec).items()},
```

Comment:
I was originally just going to suggest:
```suggestion
        "hash_to_prefix": {h: prefix for h, (_, prefix) in hashes_to_prefixes(spec).items()},
```

But then the awkward iteration made me wonder if it could be better, so with the refactor above:

```suggestion
        "hash_to_prefix": {dep.dag_hash(): dep.prefix for dep in relocatable_deps(spec)},
```

which I think is way more readable.